src/core/net/srp_server.cpp

```diff
@@ -434,17 +439,15 @@ void Server::CommitSrpUpdate(Error                    aError,
     }
     else
     {
-        LogInfo("add new host %s", aHost.GetFullName());
+        AddHost(aHost);
+        shouldFreeHost = false;
```

Comment:
Just double that changing the order and calling `AddHost()` before `for` loop is perfectly fine (adding a note to help other reviewer).

I One thing we can consider (I suggest in a future PR): I think `AddHost()` is only called from here and just logs and adds the host entry in the linked list so we can actually just do this directly here and remove the method

Reply:
Thanks for confirming the order!

I think it also makes sense to keep the `AddHost` method since there is a symmetrical `RemoveHost` method :)